### PR TITLE
blisp: unstable-2023-06-03 -> 0.0.4

### PR DESCRIPTION
**Pull request by bdd**
- Slightly change the description to be...more descriptive.
- Add a passthru test.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by moni-dz — APPROVED** [review]

File: pkgs/development/embedded/blisp/default.nix [hunk 1]
```diff
@@ -5,17 +5,22 @@
 , cmake
 , libserialport
 , pkg-config
+, testers
 , IOKit
+, blisp
 }:
 
+let
+  version = "0.0.4";
```
Comment (nagy) [inline pkgs/development/embedded/blisp/default.nix:14]: You dont need to introduce a let binding for this. Just change the version down below and use the finalAttrs pattern for stdenv.

File: pkgs/development/embedded/blisp/default.nix [hunk 2]
```diff
@@ -31,12 +36,16 @@ stdenv.mkDerivation {
     "-DBLISP_USE_SYSTEM_LIBRARIES=ON"
   ];
 
+  passthru.tests.version = testers.testVersion {
+    package = blisp;
```
Comment (nagy) [inline pkgs/development/embedded/blisp/default.nix:40]: If you use the finalAttrs pattern for stdenv you dont need to use the package name variable.

See this for an example: https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/misc/tilemaker/default.nix#L36

File: pkgs/development/embedded/blisp/default.nix (hunk 1 above)
Comment (bdd) [reply pkgs/development/embedded/blisp/default.nix:14]: Thank you.
IIUC, this way we also don't break the override functionality.

Comment (nagy) [reply pkgs/development/embedded/blisp/default.nix]: > IIUC, this way we also don't break the override functionality.

correct.

File: pkgs/development/embedded/blisp/default.nix [hunk 3]
```diff
@@ -31,12 +32,16 @@ stdenv.mkDerivation {
     "-DBLISP_USE_SYSTEM_LIBRARIES=ON"
   ];
 
+  passthru.tests.version = testers.testVersion {
+    package = finalAttrs.finalPackage;
+    command = "blisp --version";
```
Comment (nagy) [inline pkgs/development/embedded/blisp/default.nix:37]: If I am not mistaken, this is already the default ( with the `mainProgram` change below). can you check if it still work if you remove this line?

File: pkgs/development/embedded/blisp/default.nix [hunk 4]
```diff
@@ -31,12 +32,16 @@ stdenv.mkDerivation {
...
+    version = "v${finalAttrs.version}";
+  };
+
   meta = with lib; {
-    description = "ISP tool & library for Bouffalo Labs RISC-V Microcontrollers and SoCs";
+    description = "An In-System-Programming (ISP) tool & library for Bouffalo Labs RISC-V Microcontrollers and SoCs";
     license = licenses.mit;
     homepage = "https://github.com/pine64/blisp";
     maintainers = [ maintainers.fortuneteller2k ];
```
Comment (nagy) [inline pkgs/development/embedded/blisp/default.nix:45]: Can you add `mainProgram = "blisp";` ?

**Review by nagy — CHANGES_REQUESTED** [review]

File: pkgs/development/embedded/blisp/default.nix (hunk 3 above)
Comment (bdd) [reply pkgs/development/embedded/blisp/default.nix:37]: Yes. Indeed it does default to: `"${package.meta.mainProgram or package.pname or package.name} --version"`.

TIL.

File: pkgs/development/embedded/blisp/default.nix (hunk 4 above)
Comment (bdd) [reply pkgs/development/embedded/blisp/default.nix:45]: I think we cat get away without setting `mainProgram` here given the only executable the package emits is equal to the `pname`. So `nix run` will do the right thing and `testVersion` as commented above will also fall back to using `pname`.

Comment (nagy) [reply pkgs/development/embedded/blisp/default.nix:44]: That policy has recently changed #246386 .

**Review by nagy — APPROVED** [review]

**Pull request merged**